node/pkg/driver/device_connectivity/device_connectivity_helper_scsigeneric.go

```diff
@@ -579,13 +678,76 @@ func (o GetDmsPathHelperGeneric) WaitForDmToExist(volumeUuid string, volumeNguid
 			return "", err
 		}
 		dms := string(out)
-		if !strings.Contains(dms, volumeUuid) && !strings.Contains(dms, volumeNguid) {
-			err = os.ErrNotExist
-		} else {
-			return dms, nil
+		for _, identifier := range identifiers {
+			if strings.Contains(dms, identifier) {
+				return dms, nil
+			}
 		}
+		err = os.ErrNotExist
 
 		time.Sleep(time.Second * time.Duration(intervalSeconds))
 	}
 	return "", err
```

Comment:
I know it's existing code, but I don't think we need `var err error` nor `err = nil`
```suggestion

		time.Sleep(time.Second * time.Duration(intervalSeconds))
	}
	return "", os.ErrNotExist
```

Reply:
btw shouldn't `os.ErrNotExist` be instantiated (e.g. `&os.ErrNotExist{}` or something)?

Reply:
done

Reply:
thanks, let me know when `&os.ErrNotExist{}` is in as well

Reply:
I didn't understand where do you want me to add `&os.ErrNotExist{}`

Reply:
I thought maybe `return "", os.ErrNotExist` should be `return "", &os.ErrNotExist{}` - isn't that the case?

Reply:
or maybe without `&`, since the function returns the error itself and not a pointer?

Reply:
can I open a ticket for it?

Reply:
for adding `{}`? is it failing?

Reply:
it made me some problems with the unit tests

Reply:
if it doesn't work, just use a custom error

Reply:
or open a ticket, whatever you prefer

Reply:
has the [done](https://github.com/IBM/ibm-block-csi-driver/pull/491/files#r879336810) part also failed the unit tests?
if not, it would be better to keep it

Reply:
gentle reminder

Reply:
>change the behavior of the function to return a not found custom error
sounds good to me